src/apps/projinfo.cpp

```diff
@@ -1248,6 +1332,77 @@ int main(int argc, char **argv) {
         }
     }
 
+    if (listCRSSpecified) {
+        bool allow_deprecated = false;
+        std::set<AuthorityFactory::ObjectType> types;
+        auto tokens = split(listCRS, ',');
+        if (listCRS.empty()) {
+            tokens.clear();
+        }
+        for (auto token : tokens) {
+            if (ci_equal(token, "allow_deprecated")) {
+                allow_deprecated = true;
+            } else if (ci_equal(token, "geodetic")) {
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_2D_CRS);
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_3D_CRS);
+                types.insert(AuthorityFactory::ObjectType::GEOCENTRIC_CRS);
+            } else if (ci_equal(token, "geocentric")) {
+                types.insert(AuthorityFactory::ObjectType::GEOCENTRIC_CRS);
+            } else if (ci_equal(token, "geographic")) {
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_2D_CRS);
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_3D_CRS);
+            } else if (ci_equal(token, "geographic_2d")) {
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_2D_CRS);
+            } else if (ci_equal(token, "geographic_3d")) {
+                types.insert(AuthorityFactory::ObjectType::GEOGRAPHIC_3D_CRS);
+            } else if (ci_equal(token, "vertical")) {
+                types.insert(AuthorityFactory::ObjectType::VERTICAL_CRS);
+            } else if (ci_equal(token, "projected")) {
+                types.insert(AuthorityFactory::ObjectType::PROJECTED_CRS);
+            } else if (ci_equal(token, "compound")) {
+                types.insert(AuthorityFactory::ObjectType::COMPOUND_CRS);
+            } else {
+                std::cerr << "Unrecognized value for option --list-crs: " << token
+                          << std::endl;
+                usage();
+            }
+        }
+
+        auto bboxFilter = makeBboxFilter(dbContext, bboxStr, area);
+        auto allowedAuthorities(outputOpt.allowedAuthorities);
+        if (allowedAuthorities.empty()) {
+            allowedAuthorities.emplace_back(std::string());
+        }
+        for (auto auth_name : allowedAuthorities) {
+            try {
+                auto factory = AuthorityFactory::create(NN_NO_CHECK(dbContext), auth_name);
+                auto list = factory->getCRSInfoList();
+                for (const auto &info : list) {
+                    if (!allow_deprecated && info.deprecated) {
+                        continue;
+                    }
+                    if (!types.empty() && types.find(info.type) == types.end()) {
+                        continue;
+                    }
+                    if (bboxFilter) {
+                        auto crsExtent = Extent::createFromBBOX(
```

Comment:
you should check info.bbox_valid

Reply:
So the logic should be like this?:
```if (bboxFilter && !info.bbox_valid) { continue; }```
(that means that CRS without extent definition are not listed if `--bbox` or `--area` are used)

Reply:
sounds reasonable. That's more an hypothetical sense since a well behaved database should include extent for all CRS